examples/include/crypto-accelerator.p4

```diff
@@ -122,6 +128,55 @@ extern crypto_accelerator {
     void enable_encrypt<T>(in T enable_auth);
     void enable_decrypt<T>(in T enable_auth);
 
+    // crypto accelerator runs immediately and returns control flow to the current pipeline
+    // stage. The method is responsible for defining the contents of the ESP header, 
+    // calculating the payload offset and lengths, encrypting the payload appropriately and 
+    // reparsing the packet. User can decide if to proceed or reinject.
+    //
+    // Pre-conditions: The parser must have been executed prior to this extern. The packet 
+    // headers and metadata from the parser are provided as inout params.
+    // Post-conditions: The deparser will be executed prior to encapsulation, the packet 
+    // bytestream will be updated and encryption will be performed on the payload. The 
+    // packet will be reparsed and parser states updated.
+    // Side-effects: parser states will be re-evaluated if crypto has succeeded.
+    //
+    // H - inout Headers is the output of the parser block
+    // M - inout Metadata is from the parser block and shared with the control 
+    // T - in enable_auth flag enables authentication check
+    // S - in seq is the optional sequence number
+    // I - in iv is the initialization vector
+    crypto_results_e encrypt_inline<H,M,T,S,I>(packet_in pkt,
+                        inout H hdr, 
+                        inout M meta,
+                        in crypto_mode_e mode,
+                        in T enable_auth,
+                        in bit<32> spi,
+                        in S seq,
+                        in I iv);
```

Comment:
Why include all these arguments rather than reuse the `set_iv()`, `set_key()`, etc. methods above?